### ISSUE TITLE
Tags icons listed before the default entry are ignored

### ISSUE THREAD

**Issue by Guts** [squidfunk/mkdocs-material#6635]
### Bug description

Pretty tricky bug that I'm trying to catch up for months now!

Expected behavior: when using tag icons, the order of icons identifiers should not matter and every icon should be rendered, no matter if its identifier is below or above the `entry` entry.

Given this mkdocs.yml:

```yml
site_name: My Docs

plugins: 
  - tags:
      enabled: true


theme:
  name: material
  icon:
    tag:
      database: fontawesome/solid/database
      # icons above this entry are not rendered
      default: fontawesome/solid/tag
      globe: fontawesome/solid/earth-europe

extra: 
  tags:
    DuckDB: database
    GEOS: globe
    PostGIS: database
    PostgreSQL: database
    
```

This `index.md`:

```markdown
---
tags:
    - DuckDB
    - GEOS
    - PostGIS
    - PostgreSQL
---

# Welcome to MkDocs

For full documentation visit [mkdocs.org](https://www.mkdocs.org).

```

Output:

![image](https://github.com/squidfunk/mkdocs-material/assets/1596222/f1d2c089-ef6f-4bc3-aa06-284eb543b012)

Now, let's move every icon identifiers below `ðefault` entry:

```yml
[...]
theme:
  name: material
  icon:
    tag:
      # icons above this entry are not rendered
      default: fontawesome/solid/tag
      database: fontawesome/solid/database
      globe: fontawesome/solid/earth-europe
[...]
```

Output:

![image](https://github.com/squidfunk/mkdocs-material/assets/1596222/1f8d121d-3170-4373-8f2f-15b54f7c5ea7)

:partying_face: every defined icons are rendered!

### Related links

- [Tags icon](https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/?h=icon#tag-icons-and-identifiers)

### Reproduction

[9.5.3+insiders.4.49.2-plugin_tags_icons_order.zip](https://github.com/squidfunk/mkdocs-material/files/13894047/9.5.3%2Binsiders.4.49.2-plugin_tags_icons_order.zip)

### Steps to reproduce

1. Create a page with some tags
1. In `theme.icon.tag`, add some icons above and below the `default` entry
1. In `extra.tags`, match between tags and icons
1. `mkdocs serve` --> every tag whose icon entry is defined above `default` does not have its own icon rendered

### Browser

Chrome, Firefox

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in e68e1330a.

**Comment by squidfunk** [comment]
Released as part of [9.5.4](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.4).
